lib/ui/experiments/compositing_3d.dart

```diff
@@ -0,0 +1,884 @@
+    _addPicture(offset.dx, offset.dy, picture, hints);
+  }
+
+  @FfiNative<Void Function(Pointer<Void>, Double, Double, Pointer<Void>, Int32)>('SceneBuilder::addPicture')
+  external void _addPicture(double dx, double dy, Picture picture, int hints);
+
+  /// Adds a backend texture to the scene.
+  ///
+  /// The texture is scaled to the given size and rasterized at the given offset.
+  ///
+  /// If `freeze` is true the texture that is added to the scene will not
+  /// be updated with new frames. `freeze` is used when resizing an embedded
+  /// Android view: When resizing an Android view there is a short period during
+  /// which the framework cannot tell if the newest texture frame has the
+  /// previous or new size, to workaround this the framework "freezes" the
+  /// texture just before resizing the Android view and un-freezes it when it is
+  /// certain that a frame with the new size is ready.
+  void addTexture(
+    int textureId, {
+    Offset offset = Offset.zero,
+    double width = 0.0,
+    double height = 0.0,
+    bool freeze = false,
+    FilterQuality filterQuality = FilterQuality.low,
+  }) {
+    assert(offset != null, 'Offset argument was null');
+    _addTexture(offset.dx, offset.dy, width, height, textureId, freeze, filterQuality.index);
+  }
+
+  @FfiNative<Void Function(Pointer<Void>, Double, Double, Double, Double, Int64, Bool, Int32)>('SceneBuilder::addTexture', isLeaf: true)
+  external void _addTexture(double dx, double dy, double width, double height, int textureId, bool freeze, int filterQuality);
+
+  /// Adds a platform view (e.g an iOS UIView) to the scene.
+  ///
+  /// On iOS this layer splits the current output surface into two surfaces, one for the scene nodes
+  /// preceding the platform view, and one for the scene nodes following the platform view.
+  ///
+  /// ## Performance impact
+  ///
+  /// Adding an additional surface doubles the amount of graphics memory directly used by Flutter
+  /// for output buffers. Quartz might allocated extra buffers for compositing the Flutter surfaces
+  /// and the platform view.
+  ///
+  /// With a platform view in the scene, Quartz has to composite the two Flutter surfaces and the
+  /// embedded UIView. In addition to that, on iOS versions greater than 9, the Flutter frames are
+  /// synchronized with the UIView frames adding additional performance overhead.
+  ///
+  /// The `offset` argument is not used for iOS and Android.
+  void addPlatformView(
+    int viewId, {
+    Offset offset = Offset.zero,
+    double width = 0.0,
+    double height = 0.0,
+  }) {
+    assert(offset != null, 'Offset argument was null');
+    _addPlatformView(offset.dx, offset.dy, width, height, viewId);
+  }
+
+  @FfiNative<Void Function(Pointer<Void>, Double, Double, Double, Double, Int64)>('SceneBuilder::addPlatformView', isLeaf: true)
+  external void _addPlatformView(double dx, double dy, double width, double height, int viewId);
+
+  void addModelLayer(
```

Comment:
Could `experimental/compositing_3d.dart` just contain an extension method on `SceneBuilder`, and leave the rest of this file in `compositing.dart`? (An exception to the ban on extension methods seems less bad than duplicating all the code, in this case...)

Reply:
extension method no, but I could try to make this a static method. Different API on the Flutter side but that doesn't really matter for this.  Good call

Reply:
oh wait, no that won't work. Needs to be an instance method

Reply:
what if we leave the method here, but make it un-reachable without a separate static method?